### DISCUSSION TITLE
Dependencies conflict when trying to build airflow from source using constraint file

### DISCUSSION THREAD

**Discussion by SuccessMoses** [apache/airflow#43921] [General]
### What do you see as an issue?

I tried to install airflow and provider packages from source following this [documentation](https://github.com/apache/airflow/blob/main/INSTALL)

```
pip install -e ".[devel,google]" \
--constraint "https://raw.githubusercontent.com/apache/airflowconstraint-main/cinstraint-no-providers-3.9.txt"

```
this happened:

![issue](https://github.com/user-attachments/assets/e44b3399-29a6-4579-bc4a-d3606a6ad8fb)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
You are mixing first and second command. The first one when you install provider deps uses constraints, the second does not use provider extras (google) and uses "no-providers" constraints.

You merged the two commands - trying to install "no-provider" constraints with "google" provider extra. That ain't going to work.
